front/src/applications/operationalStudies/views/Scenario/components/ManageTimetableItem/Itinerary/hooks/usePathStepsMetadata.ts

```diff
@@ -0,0 +1,234 @@
+          return acc;
+        }
+        acc.push({
+          id: step.id,
+          ...step.location,
+        });
+        return acc;
+      }, []),
+    [pathSteps]
+  );
+
+  const pathStepsOperationalPoints = useTrainPathOps(infraId, strippedTrainPath, {
+    returnAllOps: true,
+  });
+
+  // 2. Since a path step containing 'operation_point' as location will have only one match
+  // with postInfraByInfraIdMatchOperationalPoints, we need to call the endpoint again with
+  // the opId corresponding uic in order to get all the possible matches and have
+  // all the secondary codes and track names
+  const opIdsWithUic = useMemo(() => {
+    const opIds = pathSteps.reduce<string[]>((acc, step) => {
+      if (!step.location) return acc;
+      if ('operational_point' in step.location) {
+        acc.push(step.location.operational_point);
+      }
+      return acc;
+    }, []);
+
+    if (opIds.length === 0) return null;
+
+    return pathStepsOperationalPoints.reduce<{ uic: number }[]>((acc, op) => {
+      const uic = op.extensions?.identifier?.uic;
+      if (uic && opIds.includes(op.id)) {
+        acc.push({ uic });
+      }
+      return acc;
+    }, []);
+  }, [pathSteps, pathStepsOperationalPoints]);
+
+  const { currentData: allOpIdsOperationalPoints } =
+    osrdEditoastApi.endpoints.postInfraByInfraIdMatchOperationalPoints.useQuery(
+      opIdsWithUic
+        ? {
+            infraId,
+            body: {
+              operational_point_references: opIdsWithUic,
+            },
+          }
+        : skipToken
+    );
+
+  // 3. Merge both operational points lists to have all possible matches
+  const allOps = useMemo(() => {
+    if (
+      !allOpIdsOperationalPoints?.related_operational_points ||
+      allOpIdsOperationalPoints.related_operational_points.length === 0
+    ) {
+      return pathStepsOperationalPoints;
+    }
+    return [
+      ...pathStepsOperationalPoints,
+      ...allOpIdsOperationalPoints.related_operational_points.flat(),
+    ];
+  }, [pathStepsOperationalPoints, allOpIdsOperationalPoints]);
+
+  useEffect(() => {
+    const fetchAndSetMetadata = async () => {
+      // 4. Get all track ids of all matched operational points to get all tracks metadata
+      const matchedTrackIds = new Set<string>();
+      allOps.forEach((op) => {
+        op.parts.forEach((part) => {
+          matchedTrackIds.add(part.track);
+        });
+      });
+      const allTrackIds = Array.from(matchedTrackIds);
+
+      // Add requested points track ids
+      pathSteps.forEach(
+        (step) => step.location && 'track' in step.location && allTrackIds.push(step.location.track)
+      );
+
+      const trackSectionsById = await getTrackSectionsByIds(allTrackIds);
+
+      // 5. Loop of the path steps to build the metadata map
+      const newPathStepsMetadataById = new Map<string, PathStepMetadata>();
+
+      pathSteps.map(async (pathStep) => {
+        if (pathStep.location && 'track' in pathStep.location) {
```

Comment:
Can you add an early return if location is undefined ? 
```suggestion
        const { location } = pathStep;
        if (!location) {
          newPathStepsMetadataById.set(pathStep.id, { isInvalid: true });
          return;
        }

        if ('track' in location) {
```

Reply:
Done